crates/wash-lib/src/start/github/api.rs

```diff
@@ -0,0 +1,290 @@
+    let main_releases = releases
+        .into_iter()
+        .filter(|release| match &release.get_main_artifact_release() {
+            Some(version) => {
+                after_version.major == version.major && after_version.minor == version.minor
+            }
+            None => false,
+        })
+        .collect::<Vec<GitHubRelease>>();
+    Ok(main_releases)
+}
+
+pub async fn new_patch_version_of_after_string(
+    owner: &str,
+    repo: &str,
+    after_version: &str,
+) -> Result<Option<Version>, Error> {
+    let after_version = after_version.strip_prefix('v').unwrap_or(after_version);
+    let after_version = match Version::parse(after_version) {
+        Ok(v) => v,
+        Err(_) => return Ok(None),
+    };
+    return match new_patch_releases_of_after(owner, repo, &after_version).await {
+        Ok(patches) => match patches.first() {
+            Some(patch) => Ok(patch.get_main_artifact_release()),
+            None => Ok(None),
+        },
+        _ => Ok(None),
+    };
+}
+
+/// GitHubRelease represents the necessary fields to determine wadm and/or wasmCloud
+/// GitHub release (https://developer.github.com/v3/repos/releases/) object
+/// has new patch version available. The fields are based on the response from the
+/// response schema from the docs.
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct GitHubRelease {
+    pub tag_name: String,
+    pub name: String,
+    #[serde(with = "github_date_format")]
+    pub published_at: DateTimeUtc,
+    pub draft: bool,
+    pub prerelease: bool,
+}
+impl PartialEq for GitHubRelease {
+    fn eq(&self, other: &Self) -> bool {
+        self.tag_name == other.tag_name
+    }
+}
+
+impl GitHubRelease {
+    pub fn get_main_artifact_release(&self) -> Option<semver::Version> {
+        match self.tag_name.strip_prefix('v') {
+            Some(v) => match semver::Version::parse(v) {
+                Ok(v) => {
+                    if self.draft || self.prerelease {
+                        None
+                    } else {
+                        Some(v)
+                    }
+                }
+                Err(_) => None,
+            },
+            None => None,
+        }
+    }
+}
+
+/// Returns the URL to fetch the latest release from the GitHub repository.
+/// doc: https://developer.github.com/v3/repos/releases/#get-the-latest-release
+fn format_latest_releases_url(owner: &str, repo: &str, page: u32) -> String {
+    format!(
+        "https://api.github.com/repos/{}/{}/releases?page={}&per_page={}",
+        owner, repo, page, GITHUB_PER_PAGE
+    )
+}
+
+async fn fetch_latest_releases(
+    owner: &str,
+    repo: &str,
+    latest_interested: &semver::Version,
+) -> Result<Vec<GitHubRelease>, reqwest::Error> {
+    let client = reqwest::ClientBuilder::default()
+        .user_agent(VERSION_FETCHER_CLIENT_USER_AGENT)
+        .with_native_certificates()
+        .build()
+        .expect("failed to build HTTP client");
```

Comment:
@joonas any concerns here w/ not providing the ability to give custom CA certs? Am I correct in thinking that this goes to github.com, so it shouldn't be an issue?

Reply:
I think certs shouldn't be a problem, but we do need to make sure that proxy variables are/continue to be supported so that end-users inside of limited-connectivity networks can continue to use the provided proxies.

Reply:
Wowzy, nice catch. Reused the already existing function which supplied the optional envvars.